code/__HELPERS/unsorted.dm

```diff
@@ -1654,13 +1654,13 @@ Game Mode config tags:
 		else
 			to_chat(C, msg)
 
-/proc/generic_projectile_fire(var/atom/target, var/atom/source, var/obj/item/projectile/projectile, var/shot_sound)
+/proc/generic_projectile_fire(var/atom/target, var/atom/source, var/projectile_type, var/shot_sound)
 	var/turf/T = get_turf(source)
 	var/turf/U = get_turf(target)
 	if (!T || !U)
 		return
 	var/obj/item/projectile/A
-	A = new projectile(T)
+	A = new projectile_type(T)
```

Comment:
This feels like it could get heretical in the future. Why the typecast change?

Reply:
The proc is meant to take a type as argument, not an atom, every instance of it being use sends a type.